### ISSUE TITLE
The state - loadCounter  is out of sync 

### ISSUE THREAD

**Issue by xipeter** [andrew1007/react-on-images-loaded#31]
https://github.com/andrew1007/react-on-images-loaded/blob/e52de29adc6b5590d84ee3783d97d9ff113b33de/src/OnImagesLoaded.tsx#L180

So, when there're multiple images, the state - loadCounter won't keep consistent between different images, say I have four images, in some cases,  say we have 4 images, the state loadCounter would be 2 rather than 4, because when the 3rd, and 4th image got loaded, the state is 1 at the moment of reading the value, any thoughts?

**Comment by andrew1007** [comment]
Yes, there can be an asynchronicity issue. If the component is converted to use hooks, then there can be a guarantee that the counter will stay in sync
